### PR TITLE
feat: [CDS-52579]: Aws Lambda Instance Sync NG Changes

### PR DESCRIPTION
**Pull request by piyush-bhuwalka**
## Describe your changes

Instance Sync changes for Aws lambda Swimlane in NG 

Screenshot 2023-03-01 at 12 48 41 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)

### CODE REVIEW

**Review by rohitk-h — CHANGES_REQUESTED** [review]
> Reviewed 63 of 81 files at r2, 18 of 18 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 81 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`126-instance/src/main/java/io/harness/entities/instanceinfo/AwsLambdaInstanceInfo.java` line 28 at r3](https://reviewable.io/reviews/harness/harness-core/43735#-NNs0M_s4y6vawQCF4pU:-NNs0M_s4y6vawQCF4pV:b-f768k7) ([raw file](https://github.com/harness/harness-core/blob/8e03593603ee80a6e2458b9c361620ed057ae4c5/126-instance/src/main/java/io/harness/entities/instanceinfo/AwsLambdaInstanceInfo.java#L28)):*
> ```Java
>   @NotNull private String version;
>   @NotNull private String functionName;
>   @NotNull private String project;
> ```

Lambda doesn't have project construct, can we remove? 
___
*[`260-delegate/src/main/java/io/harness/perpetualtask/AwsLambdaInstanceSyncPerpetualTaskExecutorNg.java` line 73 at r3](https://reviewable.io/reviews/harness/harness-core/43735#-NNs1WfG1a_ExlMD65AJ:-NNs1WfG1a_ExlMD65AK:b-drn39t) ([raw file](https://github.com/harness/harness-core/blob/8e03593603ee80a6e2458b9c361620ed057ae4c5/260-delegate/src/main/java/io/harness/perpetualtask/AwsLambdaInstanceSyncPerpetualTaskExecutorNg.java#L73)):*
> ```Java
>                                                        .collect(Collectors.toList());
> 
>     log.info("Aws Lambda Function sync nInstances: {}, task id: {}",
> ```

typo: instances
___
*[`260-delegate/src/main/java/io/harness/perpetualtask/AwsLambdaInstanceSyncPerpetualTaskExecutorNg.java` line 85 at r3](https://reviewable.io/reviews/harness/harness-core/43735#-NNs1_bw-dNsst49iQA0:-NNs1_bw-dNsst49iQA1:bk9qjap) ([raw file](https://github.com/harness/harness-core/blob/8e03593603ee80a6e2458b9c361620ed057ae4c5/260-delegate/src/main/java/io/harness/perpetualtask/AwsLambdaInstanceSyncPerpetualTaskExecutorNg.java#L85)):*
> ```Java
>       return awsLambdaTaskHelperBase.getAwsLambdaServerInstanceInfo(deploymentReleaseData);
>     } catch (Exception ex) {
>       log.warn("Unable to get google cloud function server instance", ex);
> ```

Update logger to aws lambda
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTaskHelper.java` line 28 at r3](https://reviewable.io/reviews/harness/harness-core/43735#-NNs2ARHA_FzFzTTNmRH:-NNs2ARHA_FzFzTTNmRI:b-brafm4) ([raw file](https://github.com/harness/harness-core/blob/8e03593603ee80a6e2458b9c361620ed057ae4c5/930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTaskHelper.java#L28)):*
> ```Java
> import com.amazonaws.services.ecs.model.ClusterNotFoundException;
> import com.amazonaws.services.ecs.model.ServiceNotFoundException;
> import com.amazonaws.services.lambda.AWSLambdaClient;
> ```

This is using the v1 version of the aws library, for v2 version the package is `software.amazon.awssdk.services.lambda.LambdaClient`
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaFunctionWithActiveVersions.java` line 10 at r3](https://reviewable.io/reviews/harness/harness-core/43735#-NNs32Nx3wU-DWANIkJo:-NNs32Nx3wU-DWANIkJp:b-2f41k7) ([raw file](https://github.com/harness/harness-core/blob/8e03593603ee80a6e2458b9c361620ed057ae4c5/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaFunctionWithActiveVersions.java#L10)):*
> ```Java
> package io.harness.delegate.task.aws.lambda;
> 
> import com.amazonaws.services.lambda.model.AliasConfiguration;
> ```

Same comment as above, we should use v2 aws lambda version

**Comment by rohitk-h** [comment]
*[`260-delegate/config-delegate.yml` line 19 at r17](https://reviewable.io/reviews/harness/harness-core/43735#-NPORXSv06DT2UGXwBl0:-NPORXSv06DT2UGXwBl1:bsck18e) ([raw file](https://github.com/harness/harness-core/blob/701219e55b76e9b90b59429b5e99bb1ed8f54831/260-delegate/config-delegate.yml#L19)):*
> ```YAML
> clientToolsDownloadDisabled: false
> installClientToolsInBackground: true
> isLocalNgDelegate: true
> ```

This file shouldn't be committed

**Review by rohitk-h — COMMENTED** [review]
> Reviewed 3 of 70 files at r4, 21 of 49 files at r12, 14 of 75 files at r13, 50 of 68 files at r14, 30 of 30 files at r16, 6 of 7 files at r17, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 133 file(s) to review. 1 change requests. 0 of 2 approvals obtained.

**Review by rohitk-h — COMMENTED** [review]
> Reviewed 2 of 2 files at r18, 1 of 1 files at r19, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 133 file(s) to review. 1 change requests. 0 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by rohitk-h — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 133 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by piyush-bhuwalka** [comment]
trigger ut3

**Comment by piyush-bhuwalka** [comment]
trigger ut5

**Comment by piyush-bhuwalka** [comment]
trigger ut6

**Comment by piyush-bhuwalka** [comment]
trigger ut7

**Comment by piyush-bhuwalka** [comment]
trigger ut8

**Comment by piyush-bhuwalka** [comment]
trigger ut9

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Review by rohitk-h — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 133 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by rohitk-h — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 133 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Review by rohitk-h — APPROVED** [review]
> Reviewed 3 of 3 files at r20, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 133 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by rohitk-h — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 133 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

File: 126-instance/src/main/java/io/harness/entities/instanceinfo/AwsLambdaInstanceInfo.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import lombok.EqualsAndHashCode;
+
+@OwnedBy(HarnessTeam.CDP)
+@Data
+@Builder
+@EqualsAndHashCode(callSuper = true)
+public class AwsLambdaInstanceInfo extends InstanceInfo {
+  @NotNull private String version;
+  @NotNull private String functionName;
+  @NotNull private String project;
```
Comment (alluvamsi-harness) [inline 126-instance/src/main/java/io/harness/entities/instanceinfo/AwsLambdaInstanceInfo.java:29]: can we remove project?

**Review by alluvamsi-harness — CHANGES_REQUESTED** [review]

**Comment by rohitk-h** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaTaskHelper.java` line 266 at r17](https://reviewable.io/reviews/harness/harness-core/43735#-NPUFMdV6YIYEJQbdd3z:-NPUFMdV6YIYEJQbdd4-:b-8ej4qk) ([raw file](https://github.com/harness/harness-core/blob/701219e55b76e9b90b59429b5e99bb1ed8f54831/930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaTaskHelper.java#L266)):*
> ```Java
>     }
> 
>     UpdateFunctionCodeResponse updateFunctionCodeResponseDryRun =
> ```

Is this intentional? We needed this check to ensure we only update if there is a change

File: 126-instance/src/main/java/io/harness/entities/instanceinfo/AwsLambdaInstanceInfo.java (hunk 1 above)
Comment (sainath-b) [reply 126-instance/src/main/java/io/harness/entities/instanceinfo/AwsLambdaInstanceInfo.java:29]: @piyush-bhuwalka please remove project

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/instancesync/info/AwsLambdaServerInstanceInfo.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  private String description;
+  private String runtime;
+  private String handler;
+  private String infrastructureKey;
+  private String source;
+  private Date updatedTime;
+  private Integer memorySize;
+  private String artifactId;
+
+  // todo
```
Comment (sainath-b) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/instancesync/info/AwsLambdaServerInstanceInfo.java:44]: is it still in todo state?

**Review by sainath-b — CHANGES_REQUESTED** [review]
Please address comments

File: 126-instance/src/main/java/io/harness/entities/instanceinfo/AwsLambdaInstanceInfo.java (hunk 1 above)
Comment (piyush-bhuwalka) [reply 126-instance/src/main/java/io/harness/entities/instanceinfo/AwsLambdaInstanceInfo.java]: Done

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/instancesync/info/AwsLambdaServerInstanceInfo.java (hunk 2 above)
Comment (piyush-bhuwalka) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/instancesync/info/AwsLambdaServerInstanceInfo.java]: Removed , we dont need it

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaTaskHelper.java` line 266 at r17](https://reviewable.io/reviews/harness/harness-core/43735#-NPUFMdV6YIYEJQbdd3z:-NPVVM5b5hI3JOaf23mJ:b-3a4h1m) ([raw file](https://github.com/harness/harness-core/blob/701219e55b76e9b90b59429b5e99bb1ed8f54831/930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaTaskHelper.java#L266)):*
<details><summary><i>Previously, rohitk-h (Rohit Karelia) wrote…</i></summary><blockquote>

Is this intentional? We needed this check to ensure we only update if there is a change
</blockquote></details>

Updated again

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 15 of 82 files at r2, 3 of 70 files at r4, 20 of 49 files at r12, 14 of 75 files at r13, 50 of 68 files at r14, 20 of 30 files at r16, 4 of 7 files at r17, 1 of 2 files at r18, 1 of 1 files at r19, 2 of 3 files at r20, 2 of 2 files at r21, 1 of 1 files at r22, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 3 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by sainath-b — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 3 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

**Review by alluvamsi-harness — COMMENTED** [review]
> Reviewed 15 of 82 files at r2, 3 of 70 files at r4, 20 of 49 files at r12, 14 of 75 files at r13, 50 of 68 files at r14, 20 of 30 files at r16, 4 of 7 files at r17, 1 of 2 files at r18, 1 of 1 files at r19, 2 of 3 files at r20, 2 of 2 files at r21, 1 of 1 files at r22, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by alluvamsi-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: 1. unresolved discussions. Waiting on team(s): del-team.

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 15 of 82 files at r2, 3 of 70 files at r4, 20 of 49 files at r12, 14 of 75 files at r13, 50 of 68 files at r14, 20 of 30 files at r16, 4 of 7 files at r17, 1 of 2 files at r18, 1 of 1 files at r19, 2 of 3 files at r20, 2 of 2 files at r21, 1 of 1 files at r22, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43735)* status: Waiting on team(s): del-team.

**Pull request merged**